var/spack/repos/builtin/packages/grass/package.py

```diff
@@ -0,0 +1,202 @@
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Grass(AutotoolsPackage):
+    """GRASS GIS (Geographic Resources Analysis Support System), is a free
+       and open source Geographic Information System (GIS) software suite
+       used for geospatial data management and analysis, image processing,
+       graphics and maps production, spatial modeling, and visualization."""
+
+    homepage = "http://grass.osgeo.org"
+    url      = "https://grass.osgeo.org/grass74/source/grass-7.4.1.tar.gz"
+
+    version('7.4.1',    'bf7add62cbeb05a3ed5ad832344ba524')
+
+    variant('cxx',       default=True,  description='Add c++ functionality')
+    variant('tiff',      default=True,  description='Add TIFF functionality')
+    variant('png',       default=True,  description='Add PNG functionality')
+    variant('postgres',  default=False, description='Add PostgreSQL functionality')
+    variant('mysql',     default=False, description='Add MySQL functionality')
+    variant('sqlite',    default=True,  description='Add SQLite functionality')
+    variant('opengl',    default=True,  description='Add OpenGL functionality')
+    variant('fftw',      default=True,  description='Add FFTW functionality')
+    variant('blas',      default=False, description='Add BLAS functionality')
+    variant('lapack',    default=False, description='Add LAPACK functionality')
+    variant('cairo',     default=True,  description='Add Cairo functionality')
+    variant('freetype',  default=True,  description='Add FreeType functionality')
+    variant('readline',  default=False, description='Add Readline functionality')
+    variant('regex',     default=True,  description='Add regex functionality')
+    variant('pthread',   default=False, description='Add POSIX threads functionality')
+    variant('openmp',    default=False, description='Add OpenMP functionality')
+    variant('opencl',    default=False, description='Add OpenCL functionality')
+    variant('bzlib',     default=False, description='Add BZIP2 functionality')
+    variant('netcdf',    default=False, description='Enable NetCDF support')
+    variant('geos',      default=False, description='Geometry Engine for v.buffer')
+
+    # required components
+    depends_on('gmake@3.8.1:', type='build')
+    depends_on('zlib')
+    depends_on('flex', type='build')
+    depends_on('bison', type='build')
+    depends_on('proj')
+    depends_on('gdal')
+    depends_on('python@2.7:2.9')
```

Comment:
`type=('build', 'run')`?

Reply:
Yes, for python.  Shouldn't the proj/gdal be build,link,run?  Also, should this be named grass7 to be more consistent with its r routine r-rgrass7?

Reply:
Depends on how `proj`/`gdal` are used by the package.

Don't worry about the R package, we don't want several `grass` packages if they are all the same.

Reply:
I have found that if you do not include the link for routines with pkgconfig directories, it breaks the chain.  See libxfixes and libxi changes in 8450.